### PR TITLE
refactor: Reformatted and Refactored some files to make the code better and clean.

### PR DESCRIPTION
**Pull request by Sai-Suraj-27**
# PR Description 
Reformatted and Refactored files for better readability and according to latest Python standards. Used `f-strings`, merged nested `if` conditions, and few more small changes to make the code better and clean.

## Checklist 
- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you follow the steps we provided?

### Socials: 
https://twitter.com/Sai_Suraj_27

### CODE REVIEW

**Review by vedpatwardhan — COMMENTED** [review]
I'll leave this with @KareemMAX to look into as it's formatting related. Thanks @Sai-Suraj-27 😄

**Review by KareemMAX — APPROVED** [review]
LGTM! great work @Sai-Suraj-27

File: ivy/__init__.py [hunk 1]
```diff
@@ -1407,11 +1407,11 @@ def cast_data_types(val=True):
 }
 
 # TODO: change when it's not the default mode anymore
-promotion_table = {
-    **array_api_promotion_table,
```
Comment (HaiderSultanArc) [inline ivy/__init__.py:1411]: This causes an error that breaks ivy. 😄
I don't know why CI didn't pick up this!

Comment (HaiderSultanArc) [reply ivy/__init__.py:1411]: I'll push a fix for this no need to worry 👍

Comment (Sai-Suraj-27) [reply ivy/__init__.py:1411]: > I'll push a fix for this no need to worry 👍

Sorry for the mistake, Thank you for fixing it.

**Pull request merged**